x-pack/plugins/security_solution/server/lib/detection_engine/signals/threshold/types.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  AggregationsAggregationContainer,
+  AggregationsCardinalityAggregate,
+  AggregationsCompositeAggregation,
+  AggregationsCompositeBucket,
+  AggregationsMaxAggregate,
+  AggregationsMinAggregate,
+  AggregationsMultiBucketAggregateBase,
+  AggregationsStringTermsBucket,
+  AggregationsTermsAggregation,
+} from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+export interface ThresholdLeafAggregates {
+  max_timestamp: AggregationsMaxAggregate;
+  min_timestamp: AggregationsMinAggregate;
+  cardinality_count?: AggregationsCardinalityAggregate;
+}
+
+export type ThresholdTermsBucket = AggregationsStringTermsBucket & ThresholdLeafAggregates;
+export type ThresholdCompositeBucket = AggregationsCompositeBucket & ThresholdLeafAggregates;
```

Comment:
These types can be autogenerated from the request aggregation using the `ESSearchResponse` type in core - https://github.com/elastic/kibana/blob/main/src/core/types/elasticsearch/index.ts#L26. I've had good success with it in the new terms rule type POC [here](https://github.com/elastic/kibana/pull/131010/files#diff-c7e9adc0c07ff73186167810d9544efcdd33e4b965a88db8488f54558792d9c6R12) to generate the aggregation response type from the aggregation request. It makes it much easier to modify the aggregation requests.